src/main/java/com/ldtteam/blockui/views/ScrollingListContainer.java

```diff
@@ -1,19 +1,42 @@
 package com.ldtteam.blockui.views;
 
+import com.ldtteam.blockui.Alignment;
 import com.ldtteam.blockui.Loader;
 import com.ldtteam.blockui.Pane;
 import com.ldtteam.blockui.PaneParams;
+import com.ldtteam.blockui.controls.Text;
 import com.ldtteam.blockui.util.SafeError;
 import com.ldtteam.blockui.util.records.SizeI;
 import com.ldtteam.blockui.views.ScrollingList.DataProvider;
+import net.minecraft.network.chat.MutableComponent;
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
+import java.util.List;
+
+import static com.ldtteam.blockui.controls.AbstractTextElement.DEFAULT_TEXT_COLOR;
+import static com.ldtteam.blockui.controls.AbstractTextElement.DEFAULT_TEXT_SCALE;
+
 /**
  * A Blockout pane that contains a scrolling line of other panes.
  */
 public class ScrollingListContainer extends ScrollingContainer
 {
+    /**
+     * The current empty text values.
+     */
+    private List<MutableComponent> emptyText;
+
+    /**
+     * The current empty text color for the {@link ScrollingListContainer#emptyText}.
+     */
+    private int emptyTextColor = DEFAULT_TEXT_COLOR;
+
+    /**
+     * The current empty text scale for the {@link ScrollingListContainer#emptyText}.
+     */
+    private double emptyTextScale = DEFAULT_TEXT_SCALE;
```

Comment:
Honestly i would simplify the code to:
- remove param fields, ie. keep the text field only
- mark the text field @ nullable
- set text field if any "empty" param is set and keep the instance forever - ie. dont set it to null and just add/remove from childs as needed (= the instance would get created only if requested by code/xml)